### DISCUSSION TITLE
the nozzle plunges into the bed

### DISCUSSION THREAD

**Discussion by pyrzzz** [Jyers/Marlin#914] [General Discussion]
I am using trianglelab 3d touch. sometimes it starts printing normally, but sometimes the nozzle does not stop before printing and dives into the bed. I did the wiring as shown in smith3d. Can you help me fix the problem please. thanks.

**Comment by pyrzzz** [comment]
Doesn't self test when I turn the printer off and on, it just stays

**Reply by gothcha** [reply]
If the self test fails on power up it is a hardware or wiring fault, not firmware. But I have no experience with the 3D touch.

**Reply by pyrzzz** [reply]
I mean by the test, when you turn it off and on, it does not take all axes to 0. bl touch tests itself twice, there is no problem.

**Reply by gothcha** [reply]
OK. On power up the axis do not home automatically.  See #192 for thread with lots of details of probe failure.

**Reply by gothcha** [reply]
@pyrzzz It is best to use this thread as #192 has become too long to follow.

Yes you can use Jyers but you need to compile your own firmware as the pre-compiled do not have this feature.
uncomment this line
//#define Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN

The other alternative is to fit the 10K pull-up resistor which appears to fix the issue for most users, providing it is not a probe reliability issue (sticking etc).

**Reply by pyrzzz** [reply]
I apologize to you very much, I don't know how to compile firmware. When I start compiling, do I need to copy the line you shared to the relevant place? please forgive me for tiring you

**Reply by gothcha** [reply]
Which firmware version do you need? 4.2.2? UBL? I can compile a version for you if this would help.  I cannot test it however

**Reply by pyrzzz** [reply]
I use 4.2.2 motherboard E3V2-BLTouch-3x3-v4.2.2-v1.3.3.bin and trianglelab 3dtouch. thank you so much

**Reply by pyrzzz** [reply]
In the meantime, I removed the z endstop, the outer end of the cable is idle, but the end on the motherboard is in its place. I don't know if I should remove it

**Reply by gothcha** [reply]
OK I will compile this vesrion. To confirm you are using a cable with the 2 separate connectors in the BLTouch port and the Zmin port?

**Reply by pyrzzz** [reply]
I did not disconnect the z endstop cable from the motherboard, I told you so. one end is idle and the other is connected to the motherboard. I wonder if it is parasitic and I wanted to say if I might be having this problem

**Reply by gothcha** [reply]
I understand but to use Zmin port for probe you must connect the right most 2 wires from the 3DTouch to the Zmin socket. To do this you will need to remove the wires/pins from the existing 5pin plug and remove the wires from the original Zmin plug and use it to house the 2wires from the 3DTouch if this makes sense. There are lots of tutorials for this.

**Reply by pyrzzz** [reply]
I get it, you say I have to wiring this way.

https://i1.wp.com/smith3d.com/wp-content/uploads/2021/01/zswitchstop.jpg?resize=980%2C546&ssl=1

**Reply by gothcha** [reply]
Exactly I was just going to send you the same photo!
This should make it more reliable as the BLTouch port was badly designed by Creality and didn't include the critical pull-up.

**Reply by gothcha** [reply]
Here is the new firmware, I hope this fixes your issues.

[E3V2-3DT-3x3-v4.2.2-v1.3.3-Zmin.bin.zip](https://github.com/Jyers/Marlin/files/6560556/E3V2-3DT-3x3-v4.2.2-v1.3.3-Zmin.bin.zip)

**Reply by pyrzzz** [reply]
Thank you very much, I will use it and report the result here.

**Reply by pyrzzz** [reply]
I haven't had any problems so far, for that I am grateful to you.

**Reply by gothcha** [reply]
Excellent glad to be able to help- happy printing!

**Comment by asychev** [comment]
I have Trianglelab 3DTouch and use precompiled firmware - literally zero issues. I do believe the problem might be with connections somewhere. I soldered everything and use proper JST connector to attach cable to the motherboard - included Dupont cables are no go. The example wiring above is not required, stock connector for BLTouch works without any problems.

**Reply by xantrk** [reply]
When I changed my fan duct and had to use a mount that is further away from hotend fan, my failed probes went away. I believe there is interference caused by fans. I also have TriangleLabs version

**Reply by gothcha** [reply]
@asychev @xantrk I and others also use the BLTouch with the 5 pin connector without issues, however the prime cause of the unreliable operation that many experience has been shown to be fundamentally due to the lack of the appropriate pull up resistor on the port. As a result noise etc on the line or variability in the internal weak pull-up  can generate false probe readings.

**Comment by dkustec** [comment]
Just to confirm:
- after rewiring to a Z-stop switch; BLTouch works flawlessly, without any failed probe (Antclabs OG v3.1). Few days of constant printing and zero probing failed. 
(Ender-3V2 with 4.2.2 board; with "A" marking on a uSD card shield).
